kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -294,13 +301,25 @@ namespace Kratos
         array_1d<double,3> avg_pt;
         array_1d<double,3> avg_extra_geom_normal;
         std::vector<array_1d<double,3> > aux_pts;
+        std::vector<array_1d<double,3> > aux_avg_pts;
 
         // Check which edges are intersected
         for (std::size_t i_edge = 0; i_edge < mNumEdges; ++i_edge){
             avg_pt = ZeroVector(3);
             avg_extra_geom_normal = ZeroVector(3);
             aux_pts.clear();
 
+            const auto repeated_point_check = [&] (array_1d<double,3>& rIntPt, std::vector<array_1d<double,3>>&  rAuxPts) {
+                // Check if there is a close intersection (repeated intersection point)
+                for (auto aux_pt : rAuxPts){
+                        const double aux_dist = norm_2(rIntPt - aux_pt);
+                        const double tol_edge = 1e-2*norm_2(rEdgesContainer[i_edge][0] - rEdgesContainer[i_edge][1]);
+                        if (aux_dist < tol_edge){
+                            return true;
+                    }
+                }
+                return false;
+            };
```

Comment:
Why creating a lambda for this and not a method?

Reply:
This "repeated_point_check" is used more than once but all usages are contained in the same method. Since the function conisists of a reduced number of lines and it is contained in the same method, a lambda seems cleaner to me as it reduces the amount methods defined in the header.

Just my opinion, if it is a bad practice I have nothing against defining a method instead.

Reply:
The point is that you end up with a super large function. I'd create a private method.

Reply:
Ok, done!